src/Render/FriendicaSmarty.php

```diff
@@ -22,6 +22,7 @@
 namespace Friendica\Render;
 
 use Smarty;
+use Friendica\DI;
```

Comment:
Fortunately, this is a class with a constructor. Please add the arguments at `__constructor()` and directly use the Dependency injection

Reply:
Or even better add `string $workdir` at the constructor and avoid a dependency onto a complex structure/class by delegate it to the caller :-)

Reply:
last commit:
`$work_dir` is get from config in `FriendicaSmartyEngine` and passed to the constructor of `FriendicaSmarty`.

The call to `setUseSubDirs()` that was in `FriendicaSmartyEngine` constructor has been moved into  `FriendicaSmarty`, with all other smarty setup functions.